src/passes/OptimizeInstructions.cpp

```diff
@@ -133,6 +133,69 @@ struct LocalScanner : PostWalker<LocalScanner> {
   }
 };
 
+// perform some final optimizations
+struct FinalOptimizer
+  : public WalkerPass<
+      PostWalker<FinalOptimizer, UnifiedExpressionVisitor<FinalOptimizer>>> {
+  const PassOptions& passOptions;
+
+  FinalOptimizer(const PassOptions& passOptions) : passOptions(passOptions) {}
+
+  void visitExpression(Expression* curr) {
```

Comment:
Oh, and this can be `visitBinary(Binary* curr)` as that's the only thing we care about.

Reply:
Done!